src/coreclr/vm/gcheaputilities.cpp

```diff
@@ -202,6 +202,7 @@ HRESULT LoadAndInitializeGC(LPWSTR standaloneGcLocation)
     GC_VersionInfoFunction versionInfo = (GC_VersionInfoFunction)GetProcAddress(hMod, "GC_VersionInfo");
     if (!versionInfo)
     {
+        delete gcToClr;
```

Comment:
It seems we could move the gcToClr allocation after all of these checks instead.